docs/design/storage-node-automatic-updater.md

```diff
@@ -0,0 +1,100 @@
+# Storage Node Automatic Updater
+
+## Abstract
+
+Automatic Updater is a process separate from the Storage Node that automatically downloads the latest Storage Node binary to replace the current Storage Node binary.
+
+## Background
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date.
+If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.s
+Currently we are using Docker for automatic updates but we are migrating away from docker so we need to write out own automatic update utility for Storage Nodes.
+
+## Design
+
+* Contacts Version Server to determine if Storage Node needs to be updated.
+* Downloads latest Storage Node binary.
+* Replaces current Storage Node binary with latest Storage Node binary.
+* Kills current Storage Node binary process.
+    * Storage Node Watch Dog process will start up the new Storage Node process.
```

Comment:
This may not work on Windows. Usually, the running process is locking the executable file it is started from. Therefore, it won't be possible to replace the `storagenode.exe` binary before stopping the process.

Also, when stopping the storagenode process, we should not kill it, but shut it down as gracefully as possible.

Reply:
Windows Services already have quite sophisticated watchdog features implemented:

![image](https://user-images.githubusercontent.com/468091/63014657-b22acf80-be97-11e9-9bf3-b36240624cdb.png)

So, I don't think we need to introduce new watchdog for Windows.

Reply:
With the above in mind, the automatic update for Windows may look like this:

* Contacts Version Server to determine if Storage Node needs to be updated.
* Downloads latest Storage Node binary.
* Stop the Windows Service for Storage Node.
* Replaces current Storage Node binary with latest Storage Node binary.
* Start the Windows Service for Storage Node.